wallet-new/src/Cardano/Wallet/WalletLayer/Types.hs

```diff
@@ -0,0 +1,28 @@
+module Cardano.Wallet.WalletLayer.Types
+    ( PassiveWalletLayer (..)
+    , ActiveWalletLayer (..)
+    ) where
+
+import           Universum
+
+import           Cardano.Wallet.API.V1.Types (WalletId)
+import           Pos.Wallet.Web.ClientTypes (CWalletMeta)
+
+import           Cardano.Wallet.Kernel.Diffusion (WalletDiffusion (..))
+
+
+-- | The passive wallet (data) layer. See @PassiveWallet@.
+data PassiveWalletLayer m = PassiveWalletLayer
+    { pwlGetWalletAddresses :: m [WalletId]
+    , pwlGetWalletMeta      :: WalletId -> m (Maybe CWalletMeta)
```

Comment:
@ksaric We can postpone this after we merge, but perhaps it might be a nicer pattern to port all the functions which can fail to yield something like `m (Either WalletError a)`.

Follow-up: Why are we using `CWalletMeta` here? `ClientTypes` is effectively deprecated, so we should never leak V0 stuff into the interface. What am I missing?

Reply:
> Converting all these to Either will result in the same semantics and I'm not sure what would we gain.

In the general case you would gain an explanation on why the computation failed, which can be useful to take more informative decisions (for example to show better error messages to the user).